### PR TITLE
welcome and html auth success/failure pages

### PR DESCRIPTION
**Pull request by will-weiss**
Closes #84

### CODE REVIEW

File: src/router.ts [hunk 1]
```diff
@@ -24,8 +24,11 @@ export function createRouter(withRouter: (router: Router) => Router = identity):
     .use('/v1', v1Router.routes(), v1Router.allowedMethods())
 
   // tslint:disable-next-line:no-expression-statement
-  readdirSync(process.cwd() + '/public').forEach(fileName =>
-    router.get('/' + fileName, ctx => send(ctx, `/public/${fileName}`)))
+  readdirSync(process.cwd() + '/public').forEach(fileName => {
+    const file = parse(fileName)
+    const route = file.ext === '.html' ? file.name : fileName
+    return router.get('/' + route, ctx => send(ctx, `/public/${fileName}`))
```
Comment (shachar-langer) [inline src/router.ts:30]: Are there any benefits for creating a route for each static file? Why not using `koa-static`?

File: public/auth-failure.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<p>Oh no!</p>
```
Comment (shachar-langer) [inline public/auth-failure.html:1]: I wasn't able to reach this page by entering the wrong credentials. I reached a Twitter page instead.
How can we reach this page?

File: public/welcome.html [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<head>
```
Comment (shachar-langer) [inline public/welcome.html:1]: I wasn't able to reach this page without manually going to "/welcome.html". We will route to this page in future PRs?

File: public/welcome.html [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        <svg class="twitter-logo" viewBox="0 0 400 400">
+          <path d="M153.62 301.59c94.34 0 145.94-78.16 145.94-145.94 0-2.22 0-4.43-.15-6.63A104.36 104.36 0 00325 122.47a102.38 102.38 0 01-29.46 8.07 51.47 51.47 0 0022.55-28.37 102.79 102.79 0 01-32.57 12.45 51.34 51.34 0 00-87.41 46.78A145.62 145.62 0 0192.4 107.81a51.33 51.33 0 0015.88 68.47A50.91 50.91 0 0185 169.86v.65a51.31 51.31 0 0041.15 50.28 51.21 51.21 0 01-23.16.88 51.35 51.35 0 0047.92 35.62 102.92 102.92 0 01-63.7 22 104.41 104.41 0 01-12.21-.74 145.21 145.21 0 0078.62 23" />
+        </svg>
+        Log in with Twitter
+      </a>
+    </div>
+    <p class="terms">
+      By joining, I agree to Roar's <a href="https://roar.morehumaninternet.org/tos">TOS</a> and <a href="https://roar.morehumaninternet.org/privacy">Privacy</a>.
+      <br>
+      Protected by Twitter's <a href="https://roar.morehumaninternet.org/tos">TOS</a> and <a href="https://roar.morehumaninternet.org/privacy">Privacy</a>.
```
Comment (shachar-langer) [inline public/welcome.html]: Wrong links.

File: public/welcome.html [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    <div class="sign-in-btn__container">
+      <a class="sign-in-btn" href="/v1/auth/twitter">
+        <svg class="twitter-logo" viewBox="0 0 400 400">
+          <path d="M153.62 301.59c94.34 0 145.94-78.16 145.94-145.94 0-2.22 0-4.43-.15-6.63A104.36 104.36 0 00325 122.47a102.38 102.38 0 01-29.46 8.07 51.47 51.47 0 0022.55-28.37 102.79 102.79 0 01-32.57 12.45 51.34 51.34 0 00-87.41 46.78A145.62 145.62 0 0192.4 107.81a51.33 51.33 0 0015.88 68.47A50.91 50.91 0 0185 169.86v.65a51.31 51.31 0 0041.15 50.28 51.21 51.21 0 01-23.16.88 51.35 51.35 0 0047.92 35.62 102.92 102.92 0 01-63.7 22 104.41 104.41 0 01-12.21-.74 145.21 145.21 0 0078.62 23" />
+        </svg>
+        Log in with Twitter
+      </a>
+    </div>
+    <p class="terms">
+      By joining, I agree to Roar's <a href="https://roar.morehumaninternet.org/tos">TOS</a> and <a href="https://roar.morehumaninternet.org/privacy">Privacy</a>.
```
Comment (shachar-langer) [inline public/welcome.html:92]: We should probably hide this line until we create the relevant files.

File: src/router.ts (hunk 1 above)
Comment (will-weiss) [reply src/router.ts:30]: Thanks for bringing that up. We want `GET /welcome` to be the welcome page, not `GET /welcome.html`. Took a look at the source code and it looks like they're doing pretty much what we're doing, but not sure how to configure it to strip the `.html` for koa files. This is working nicely, so let's leave it alone.

File: public/auth-failure.html (hunk 2 above)
Comment (will-weiss) [reply public/auth-failure.html:1]: Hit cancel in this view

image

File: public/welcome.html (hunk 3 above)
Comment (will-weiss) [reply public/welcome.html:1]: Ah yes, thought I had done that. [PR here](https://github.com/morehumaninternet/roar-extension/pull/168
)

File: public/welcome.html (hunk 5 above)
Comment (will-weiss) [reply public/welcome.html:92]: That's where they'll be - I think it's fine

**Review by shachar-langer — APPROVED** [review]

**Pull request merged**